src/adcs_io.c

```diff
@@ -59,6 +77,21 @@ ADCS_returnState request_uart_telemetry(uint8_t TM_ID, uint8_t* telemetry,
   return ADCS_OK;
 }
 
-ADCS_returnState request_i2c_telemetry(uint8_t* command, uint32_t length) {
+ADCS_returnState request_i2c_telemetry(uint8_t TM_ID, uint8_t *telemetry,
+                                       uint32_t length)
+{
+  uint8_t reply[length];
+
+  i2c_receive(reply, TM_ID, length);
+
+  for (int i = 0; i < length; i++)
+  {
+    *(telemetry + i) = reply[i];
+  }
+
+  // uint8_t err_reply[6];
+  // i2c_receive(err_reply, COMMS_STAT_ID, 6);
+  // uint8_t TL_err_flag = (err_reply[4] >> 3) & 1;
```

Comment:
There was something mentioned about this in the i2c telemetry formatting section of the Firmware Manual. I was unsure whether to implement this here?

Reply:
Oh, I looked at it. First, you should know that the functions that it talks about have been developed in the main handler file and basically you can call them here to check the error flag and reset it. But, my question when I went through it was that how you could send a TM request to read the communication status if you're using I2C to send the very same TM. It looks like a loop to me.

Reply:
Yeah that was my thought as well. 
By sending this TM request, how sure are we that it itself is not prone to error, right?

Reply:
Maybe this could be handled as a separate function?

Reply:
Yeah, I'm not sure. Right now, we don't have any resources other than the manual. I don't think the ADCS team would know anything about this, either. I'd say do your best and put some comments. We'd probably need to work more on these functions and do some debugging once the ADCS is ready to integrate. It might make the ceedling tests harder I guess so maybe you can use `#ifdef mocked` kind of thing for that.

Reply:
Yes, that sounds good, I will leave it in an `#ifdef mocked` for now, so as not to completely get rid of it so that it can be dealt with later.